test/performance/odp_dma_perf.c

```diff
@@ -733,8 +734,16 @@ static int run_dma_sync(test_config_t *config)
 		for (int i = 0; i < config->dma_rounds; ++i) {
 			config->test_case_api.trs_dyn_fn(config, offset, trs_lengths[i]);
 
-			if (odp_dma_transfer(config->dma_config.handle, &trs_params[i], NULL)
-			    <= 0) {
+			while (1) {
+				started = odp_dma_transfer(config->dma_config.handle,
```

Comment:
In case of sync calls, the return value is transfers completed (not started). E.g. "num" or "ret" could be better variable name than "started".

Instead of a forever loop (e.g. problematic in CI), there could be e.g. 1000 retries with odp_time_wait_ns(1000) (1 usec delay in between). Test could report in the end number of retries done, so that it is easy to see if that happens.

Reply:
Name changed and new patch (8ad80d2) added to handle max retries in v3.